crates/telio-proxy/src/proxy.rs

```diff
@@ -191,6 +227,39 @@ impl StateEgress {
 
         Ok(())
     }
+
+    fn log_error(&mut self, err: &std::io::Error) {
+        self.conn_state = Err(err.kind());
+        telio_log_error!("Unable to send. {}", err);
+    }
+
+    async fn handle_error(&mut self, err: std::io::Error) {
+        match self.conn_state {
+            Ok(()) => self.log_error(&err),
+            Err(e) if e != err.kind() => self.log_error(&err),
+            Err(_) => (),
+        }
+
+        match ErrorType::from(err) {
```

Comment:
```suggestion
     async fn handle_error(&self, err: ErrorKind) {
        telio_log_error!("Unable to send. {}", err);

        match ErrorType::from(err) {
```
`log_error` can be removed and change `handle_error` input to `ErrorKind` to simplify things

Reply:
just to add that, following suggestion above `conn_state` could either be assigned here or before this function calls